gradle.properties

```diff
@@ -64,8 +64,8 @@ versioningPluginVersion=1.1.0
 # Versions of node and npm to use during the build. If set, these versions
 # will be downloaded and used. If not set, the existing local installations will be use
 # The version of npm corresponds to the given version of node
-npmVersion=6.14.13
-nodeVersion=14.17.1
+npmVersion=8.1.2
+nodeVersion=16.13.1
```

Comment:
As I alluded to previously, starting in NPM v7 they changed how it resolves peer dependencies. This doesn't cause a problem for our initial unchanged (NPM v6) package-lock.json files, however, npm does warn about this:

<img width="631" alt="Screen Shot 2021-12-16 at 3 18 27 PM" src="https://user-images.githubusercontent.com/3926239/146466586-b56a1b04-2c74-44de-9598-b942b050b20b.png">

As a result, when I attempted to rehydrate Biologics' package-lock.json with NPM v8 I ran into a common failure mode where there is a collision in our peer dependencies (this actually resulted in an infinite loop where I has to `kill -9` the process ... but I digress):

<img width="468" alt="Screen Shot 2021-12-16 at 3 22 34 PM" src="https://user-images.githubusercontent.com/3926239/146466599-15055c00-25e0-489e-a969-32f6c9267cdb.png">

The quick solution is to always use a new flag introduced in npm v7 called `legacy-peer-deps` ([docs here](https://docs.npmjs.com/cli/v7/using-npm/config#legacy-peer-deps)) when rehydrating. Like this:

```
npm install --legacy-peer-deps
```

This is not something we will want to be doing for very long -- if at all. We may want to spend some time to see if we can untangle these dependencies by either upgrading and/or removing out-of-date dependencies. That, however, can be a bit time consuming so the flag gives us a chance to keep going and follow up on individual modules when we update them. That said, we'd need to let folks know about this new flag and when/why to use it.

Reply:
I recommend we move ahead with informing folks about the `--legacy-peer-deps` flag when it comes to rehydrating/updating packages.

Reply:
I was able to dive a bit deeper today and have several findings to report:

1. Although we can specify the `--legacy-peer-deps` flag this still produces a v2 package-lock.json file. This means that any subsequent package updates we perform within a module/package will initially result in updating the package-lock.json from v1 (npm 6) to v2 (npm 7+). Due to this behavior, I think we should consider updating all associated package-lock.json's we "actively" maintain in coordinated PRs with these changes. I don't think this needs to be _all_ packages initially as we can do follow-up updates to modules/packages we touch less often.
2. As a result of this update we'll be updating the version of Node that our unit tests run on. I've seen some failure locally after updating. We will need to coordinate PRs to fix these tests.

Reply:
Further update:

I was able to fix issues (e.g. https://github.com/webpack/webpack-cli/issues/2990) seen using `npm run start-link` by updating `@labkey/build` to use more recent versions. These changes will be coordinated. Additonally, I've updated the associated `package-lock.json` to v2 for LKB, LKSM, Inventory, @labkey/workflow, and @labkey/freezermanager. I've opened associated branches on LKB, LKSM, and Inventory so far and will do platform in the next iteration.